lib/index.js

```diff
@@ -215,30 +218,24 @@ class Mcdev {
             if (selectedTypesArr) {
                 for (const selectedType of selectedTypesArr) {
                     const [type, subType] = selectedType ? selectedType.split('-') : [];
+                    const removePathArr = [properties.directories.retrieve, cred, bu, type];
                     if (
                         type &&
                         subType &&
                         MetadataTypeInfo[type] &&
                         MetadataTypeDefinitions[type].subTypes.includes(subType)
                     ) {
                         // Clear output folder structure for selected sub-type
-                        File.removeSync(
-                            File.normalizePath([
-                                properties.directories.retrieve,
-                                cred,
-                                bu,
-                                type,
-                                subType,
-                            ])
-                        );
+                        removePathArr.push(subType);
                         retrieveTypesArr.push(selectedType);
                     } else if (type && MetadataTypeInfo[type]) {
                         // Clear output folder structure for selected type
-                        File.removeSync(
-                            File.normalizePath([properties.directories.retrieve, cred, bu, type])
-                        );
                         retrieveTypesArr.push(type);
                     }
+                    if (!key) {
+                        // dont delete directories if we are just re-retrieving a single file
+                        File.removeSync(File.normalizePath(removePathArr));
```

Comment:
given you are changing this line - please move to async if possible

Reply:
done